proc/coredump.c

```diff
@@ -0,0 +1,518 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system kernel
+ *
+ * Process coredump
+ *
+ * Copyright 2025 Phoenix Systems
+ * Author: Jakub Klimek
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "coredump.h"
+#include "arch/cpu.h"
+#include "arch/exceptions.h"
+#include "hal/exceptions.h"
+#include "log/log.h"
+#include "proc/elf.h"
+#include "proc/process.h"
+#include <stddef.h>
+
+#ifdef COREDUMP
```

Comment:
Perhaps we could add whole .c or .o to the build process instead of global ifdef?

Side note - name is kinda too generic, there's a change of collision. `PROC_` prefix might be a good idea